### PR TITLE
[Merged by Bors] - Support specifying a namespace to watch

### PR DESCRIPTION
**Pull request by maltesander**
## Description

try
```
cargo run -- run --watch-namespace test
```
or
```
export WATCH_NAMESPACE=test
cargo run -- run
```

fixes https://github.com/stackabletech/zookeeper-operator/issues/427

TODO:
- [x] switch to tag

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by lfrancke** [comment]
I'd like to review this when it's ready

**Comment by maltesander** [comment]
> I'd like to review this when it's ready

It is ready, just waiting for the operator-rs tag.

File: docs/modules/ROOT/pages/env_var_args.adoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+*Required*: false
+
+*Multiple values:* false
+
+The operator will **only** watch for resources in the provided namespace `test`:
+
+[source]
+----
+export WATCH_NAMESPACE=test
+cargo run -- run
```
Comment (lfrancke) [inline docs/modules/ROOT/pages/env_var_args.adoc:27]: I'm not sure how helpful the references to `cargo` will be to users but I don't have a great idea for something better either, ideally we'd tell people how to configure this when run from the Docker containers in Pods but I'm fine with this now.

File: docs/modules/ROOT/pages/commandline_args.adoc [hunk 2]
```diff
@@ -7,5 +7,20 @@
...
+----
 
-This file contains property definitions for the Apache Zookeeper configuration.
+=== watch-namespace
+
+*Required*: false
+
+*Multiple values:* false
+
+The operator will **only** watch for resources in the provided namespace `test`:
```
Comment (lfrancke) [inline docs/modules/ROOT/pages/commandline_args.adoc:21]: Could you please document here what the default case is? (in the env section as well please)

**Review by lfrancke — APPROVED** [review]

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**